pkg/cmd/optimizeex.go

```diff
@@ -0,0 +1,198 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/c9s/bbgo/pkg/data/tsv"
+	"github.com/c9s/bbgo/pkg/optimizer"
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+	"io"
+	"io/ioutil"
+	"os"
+	"time"
+)
+
+func init() {
+	optimizeExCmd.Flags().String("optimizer-config", "optimizer.yaml", "config file")
+	optimizeExCmd.Flags().String("name", "", "assign a name to the study")
+	optimizeExCmd.Flags().Bool("json-keep-all", false, "keep all results of trials")
+	optimizeExCmd.Flags().String("output", "output", "backtest report output directory")
+	optimizeExCmd.Flags().Bool("json", false, "print optimizer metrics in json format")
+	optimizeExCmd.Flags().Bool("tsv", false, "print optimizer metrics in csv format")
+	RootCmd.AddCommand(optimizeExCmd)
+}
+
+var optimizeExCmd = &cobra.Command{
+	Use:   "optimizeex",
+	Short: "run hyperparameter optimizer (experimental)",
+
+	// SilenceUsage is an option to silence usage when an error occurs.
+	SilenceUsage: true,
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		optimizerConfigFilename, err := cmd.Flags().GetString("optimizer-config")
+		if err != nil {
+			return err
+		}
+
+		configFile, err := cmd.Flags().GetString("config")
+		if err != nil {
+			return err
+		}
+
+		studyName, err := cmd.Flags().GetString("name")
```

Comment:
Call it session (optimization session)? because we have another concept called Study which is in the roadmap.

Reply:
Sure, let me rename it.